lib/kadm5/log.c

```diff
@@ -1206,9 +1210,9 @@ kadm5_log_rename(kadm5_server_context *context,
         ret = krb5_store_principal(sp, source);
     if (ret == 0) {
         errno = 0;
-        if (krb5_storage_write(sp, value.data, value.length) !=
-            (krb5_ssize_t)value.length)
-            ret = errno ? errno : EIO;
+        bytes = krb5_storage_write(sp, value.data, value.length);
+        if (bytes != (krb5_ssize_t)value.length)
+            ret = errno ? errno : ENOMEM;
```

Comment:
No, it's not.  However, the `ret` assignment needs to change to `ret = bytes == -1 ? errno : ENOMEM;`.